File: a2zSheet/step1_basics/CPP/math.cpp
#include<bits/stdc++.h>

using namespace std;


int countDigits(int n){
    if(n==0)return 1;
    return (int)(log10(n)+1);
}

int reverseNumber(int n){
    int num = 0;
    while(n>0){
        num = num*10 + n%10;
        n /= 10;
    }

    return num;
}

int isPal(int n){
    int num = reverseNumber(n);
    return num==n;
}

int gcd(int a,int b){
    while(a>0 and b>0){
        if(a>b)a %= b;
        else b %= a;
    }
    if(a==0)return b;
    return a;
}

int isArmStrong(int n){
    int num = 0;
    int temp = n;
    while(n>0){
        num += (n%10)*(n%10)*(n%10);
        n /= 10;
    }
    return num == temp;
    
}

vector<int> divisors(int n){
    vector<int>ans;
    for(int i=1;i*i<=n;i++){
        if(n%i==0){
            ans.push_back(i);
            if(n/i != i)ans.push_back(n/i);
        }
    }
    return ans;
}

bool isPrime(int n){
    for(int i=2;i*i<=n;i++){
        if(n%i==0){
            return false;
        }
    }
    return true;
}


int main(){
    int n;cin>>n;
    // cout<<countDigits(n);
    // cout<<reverseNumber(n);
    cout<<isArmStrong(n);

    return 0;
}

File: a2zSheet/step1_basics/GO/math.go
package main

import (
	"bufio"
	. "fmt"
	. "math"
	"os"
)

var in = bufio.NewReader(os.Stdin)
var out = bufio.NewWriter(os.Stdout)

func CountDigits(n int) int {
	if n == 0 {
		return 1
	}
	return int(Log10(float64(n)) + 1)
}

func ReverseNumber(n int) int {
	newNum := 0
	for n > 0 {
		rem := n % 10
		n /= 10
		newNum = newNum*10 + rem
	}

	return newNum
}

func IsPalNum(n int) bool {
	reversed := ReverseNumber(n)

	return n == reversed
}
func Gcd(a, b int) int {
	for a > 0 && b > 0 {
		if a > b {
			a %= b
		} else {
			b %= a
		}
	}
	if a == 0 {
		return b
	}
	return a
}

func IsArmstrong(n int) bool {
	temp := n
	newNum := 0
	for n > 0 {
		rem := n % 10
		n = n / 10
		newNum += int(Pow(float64(rem), 3))
	}
	return newNum == temp
}

func Divisors(n int) []int {
	var divisors []int

	for i := 1; i*i <= n; i++ {
		if n%i == 0 {
			divisors = append(divisors, i)
			if n/i != i {
				divisors = append(divisors, n/i)
			}
		}
	}
	return divisors
}

func IsPrime(n int) bool {

	for i := 2; i*i < n; i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func solve() {
	var n int
	Fscan(in, &n)
	// Print(CountDigits(n), "\n")
	// Print(IsPalNum(n), "\n")
	// Print(Gcd(10, 15), "\n")
	// Print(IsArmstrong(153), "\n")
	Print(IsPrime(n), "\n")

}

func main() {
	defer out.Flush()
	var Test int
	Fscanf(in, "%d\n", &Test)
	for tt := 0; tt < Test; tt++ {
		solve()
	}
}
